charts/camunda-platform-latest/templates/web-modeler/deployment-restapi.yaml

```diff
@@ -34,11 +34,13 @@ spec:
           env:
             - name: JAVA_OPTIONS
               value: "-XX:MaxRAMPercentage=80.0"
+            {{- if or (not .Values.webModeler.restapi.externalDatabase.enabled) .Values.webModeler.restapi.externalDatabase.password }}
```

Comment:
I see that in [web-modeler/secret-restapi.yaml](https://github.com/camunda/camunda-platform-helm/blob/1b1b6d8b0e651f1086abff9a65613e5032c6aa44/charts/camunda-platform-latest/templates/web-modeler/secret-restapi.yaml) there are other conditions like `.Values.webModeler.restapi.externalDatabase.existingSecret`.

The conditions in both places should be the same.
Maybe also make a named template for the condition because it's super long now.

I leave this to @jessesimpson36 to review.

Reply:
NOOOO!!!! There's no such thing as boolean functions in named templates!  

Since "include" always returns a string, the best we could do is something like

```
{{- if eq  "true"  (include "webModeler.ShouldUsePassword" .) -}}
```

This was finding 3 in my https://github.com/jessesimpson36/helm_frustration_1?tab=readme-ov-file#finding-3-include-function-always-evaluates-to-a-string-datatype

I can still review though.

Reply:
@aabouzaid ,  I wrote this PR 
https://github.com/camunda/camunda-platform-helm/pull/1992

as an attempt to deduplicate this expression, however, I still find it not as readable. I think we should discuss this further because this is a general problem I have in helm. (perhaps as a knowledge-transfer session or team meeting topic).

Reply:
yeah it's annoying that the named template return string only ... there are many useful use cases that could be better if it was supported.